### DISCUSSION TITLE
Copier Template Tester

### DISCUSSION THREAD

**Discussion by KyleKing** [copier-org/copier#874] [Show and tell]
I wanted a way to more quickly iterate on templates, so I built this MVP pre-commit and CLI tool to parametrize testing: [copier-template-tester](https://github.com/KyleKing/copier-template-tester)

I thought this would be of interest in the community, since I hadn't found anything like it. The package has a lot of boilerplate, but the core logic is only a wrapper of the `copier.run_auto` function with arguments set in a custom `ctt.toml` file

**Comment by yajo** [comment]
I see. It looks very good! The objective is to have a folder with generated templates so you can see the output, right? Nice idea.

Thanks for publishing!

**Reply by KyleKing** [reply]
Thanks!

**Reply by KyleKing** [reply]
Yeah, the other benefit is that I can run auto formatters on the output and make sure the Jinja templates have the right spacing and quotes
